components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -3260,4 +3363,193 @@ boolean areAllDistinct(List<org.wso2.carbon.identity.api.server.idp.v1.model.Pro
+                fileContent = parseIdpToYaml(identityProvider, fileNameSB);
+                break;
+            default:
+                log.warn(String.format("Unsupported file type: %s requested for export. Defaulting to YAML parsing.",
+                        fileType));
+                fileContent = parseIdpToYaml(identityProvider, fileNameSB);
+                break;
+        }
+        return new FileContent(fileNameSB.toString(), MediaType.APPLICATION_OCTET_STREAM_VALUE, fileContent);
+    }
+
+    private String parseIdpToXml(IdentityProvider identityProvider, StringBuilder fileName)
+            throws IdentityProviderManagementException {
+
+        JAXBContext jaxbContext;
+        fileName.append(XML_FILE_EXTENSION);
+        try {
+            jaxbContext = JAXBContext.newInstance(IdentityProvider.class);
+            Marshaller marshaller = jaxbContext.createMarshaller();
+            marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+            StringWriter stringWriter = new StringWriter();
+            marshaller.marshal(identityProvider, stringWriter);
+            return stringWriter.toString();
+        } catch (JAXBException e) {
+            throw new IdentityProviderManagementException(
+                    "Error when parsing identity provider to XML file.", e);
+        }
+    }
+
+    private String parseIdpToJson(IdentityProvider identityProvider, StringBuilder fileName)
+            throws IdentityProviderManagementException {
+
+        fileName.append(JSON_FILE_EXTENSION);
+        ObjectMapper objectMapper = new ObjectMapper(new JsonFactory());
+        try {
+            return objectMapper.writeValueAsString(identityProvider);
+        } catch (JsonProcessingException e) {
+            throw new IdentityProviderManagementClientException(
+                    "Error when parsing identity provider to JSON file.", e);
+        }
+    }
+
+    private String parseIdpToYaml(IdentityProvider identityProvider, StringBuilder fileName)
+            throws IdentityProviderManagementException {
+
+        fileName.append(YAML_FILE_EXTENSION);
+        Yaml yaml = new Yaml();
+        try {
+            return yaml.dump(identityProvider);
+        } catch (YAMLException e) {
+            throw new IdentityProviderManagementException(
+                    "Error when parsing identity provider to YAML file.", e);
+        }
+    }
+
+    private String doImportIDP(InputStream fileInputStream, Attachment fileDetail, boolean isAppUpdate)
+            throws IdentityProviderManagementClientException {
+
+        IdentityProvider identityProvider;
+        try {
+            String tenantDomain = ContextLoader.getTenantDomainFromContext();
+            FileContent idpFileContent = new FileContent(fileDetail.getDataHandler().getName(),
+                    fileDetail.getDataHandler().getContentType(),
+                    IOUtils.toString(fileInputStream, StandardCharsets.UTF_8.name()));
```

Comment:
Add a separate try block - check whether there is any improvement in performance.